model/src/main/kotlin/config/PackageConfiguration.kt

```diff
@@ -31,8 +33,9 @@ import org.ossreviewtoolkit.model.VcsType
 import org.ossreviewtoolkit.utils.common.replaceCredentialsInUri
 
 /**
- * A configuration for a specific package and provenance. It allows to set up [PathExclude]s and
- * [LicenseFindingCuration]s, similar to how it is done via the [RepositoryConfiguration] for projects.
+ * A configuration for a specific [Package] (or [Project]) and [Provenance] to allow for setting up [PathExclude]s and
+ * [LicenseFindingCuration]s. When used in the context of the [OrtConfiguration], the [Identifier] may only refer to
+ * [Package]s. When used in the context of a [RepositoryConfiguration], the [Identifier] may only refer to [Project]s.
```

Comment:
@fviernau and @mnonnenmacher, I'm not actually sure myself whether the last two "When" sentences are correct 😅

At least technically, [this code](https://github.com/oss-review-toolkit/ort/blob/59702b2966ee8c328c4264eac0e86c3d14b387e3/plugins/commands/evaluator/src/main/kotlin/EvaluatorCommand.kt#L294-L312) seems to just merge all package configurations from all sources, thus allowing also package configurations that refer to *projects* from any package configuration provider.

Do we really want to technically allow this? Is doping so at least discouraged?

I simply want to clarify the docs here in that regard, as it continues to confuse me that `PackageConfiguration`s (in contrast to `PackageCuration`s) can also refer to projects.

Reply:
To me it seems very surprising that Package configuration may target project.
I guess that this is a mistake? (But I wonder how such mistake can happen)

Because for projects, the `ort.yml` has dedicated properties for path excludes and license finding curations.

Reply:
The "When [..]" sentences should be dropped.

Reply:
> > I guess that this is a mistake? (where does the assumption come from?)
> 
> I comes from a constant confusion on my end that license finding _curations_ for _packages_ are defined as part of a `Package*Configurations*` class, but license finding _curations_ for _projects_ are defined as part of a `Curations` class.
> 

This again is a mis-understanding. `LicenseFindingCuration` is the class, and it is used in both, `RepositoryConfiguration` and `PackageConfiguration`.

> Anyway, I seem to have misremembered things, and as `PackageConfigurations` is not reused in `RepositoryConfiguration` (which instead has `Curations`), `PackageConfigurations` really only applied to packages, which is good.

Reply:
> This again is a mis-understanding. `LicenseFindingCuration` is the class, and it is used in both, `RepositoryConfiguration` and `PackageConfiguration`.

I believe my sentence was still correct, and does not contract yours. What I'm after is that for `PackageConfiguration`, `LicenseFindingCuration` is directly used in a top-level property. But for `RepositoryConfiguration`, it's nested in `Curations`, which on top of that combines `LicenseFindingCuration` with `PackageCuration`.